src/docfx/build/metadata/MetadataProvider.cs

```diff
@@ -41,20 +39,21 @@ public MetadataProvider(Config config)
             {
                 if (glob(file.FilePath))
                 {
-                    fileMetadata[key] = value;
+                    // Assign a JToken to a property erases line info, so clone here.
+                    // See https://github.com/JamesNK/Newtonsoft.Json/issues/2055
+                    fileMetadata[key] = JsonUtility.DeepClone(value);
                 }
             }
             JsonUtility.Merge(result, fileMetadata);
 
             if (yamlHeader != null)
             {
-                errors.AddRange(MetadataValidator.ValidateGlobalMetadata(yamlHeader));
                 JsonUtility.Merge(result, yamlHeader);
             }
 
-            // We are validating against the merged JObject so discard the validation result here.
-            var (_, obj) = JsonUtility.ToObject<T>(result);
-            return (errors, obj);
+            var (errors, metadata) = JsonUtility.ToObject<T>(result);
```

Comment:
with this change, the merged `JObject` here has the correct line info for each value, so we have a single entry for metadata validation.